CMakeLists.txt

```diff
@@ -56,26 +79,70 @@ find_package(gsi REQUIRED)
 
 add_subdirectory(baselib/regional_esg_grid.fd)
 add_subdirectory(baselib)
-add_subdirectory(mosaic)
-add_subdirectory(lightning)
-add_subdirectory(ref2tten)
-add_subdirectory(cloudanalysis)
-add_subdirectory(larccld)
-add_subdirectory(metarcld)
-add_subdirectory(update_sst)
-add_subdirectory(use_raphrrr_sfc)
-add_subdirectory(update_gvf)
-add_subdirectory(update_snow)
-add_subdirectory(update_ice)
-add_subdirectory(lakesurgery)
-add_subdirectory(gen_cs)
-add_subdirectory(adjust_soiltq)
-add_subdirectory(update_bc)
-add_subdirectory(ens_mean_recenter)
-add_subdirectory(FV3_ensmean_recenter)
-add_subdirectory(bufrsnd)
-add_subdirectory(radmon)
-add_subdirectory(rtma_minmaxtrh)
-add_subdirectory(pm)
-add_subdirectory(blending)
+if(FV3_ENSMEAN_RECENTER)
+  add_subdirectory(FV3_ensmean_recenter.fd)
+endif()
+if(ADJUST_SOILTQ)
+  add_subdirectory(adjust_soiltq.fd)
+endif()
+if(BLENDING)
+  add_subdirectory(blending.fd)
+endif()
+if(BUFRSND)
+  add_subdirectory(bufrsnd.fd)
+endif()
+if(CLOUDANALYSIS)
+  add_subdirectory(cloudanalysis.fd)
+endif()
+if(ENS_MEAN_RECENTER)
+  add_subdirectory(ens_mean_recenter.fd)
+endif()
+if(GEN_CS)
+  add_subdirectory(gen_cs.fd)
+endif()
+if(LAKESURGERY)
+  add_subdirectory(lakesurgery.fd)
+endif()
+if(LARCCLD)
+  add_subdirectory(larccld.fd)
+endif()
+if(LIGHTNING)
+  add_subdirectory(lightning.fd)
+endif()
+if(METARCLD)
+  add_subdirectory(metarcld.fd)
+endif()
+if(MOSAIC)
+  add_subdirectory(mosaic.fd)
+endif()
+if(PM)
+  add_subdirectory(pm.fd)
+endif()
+if(RADMON)
+  add_subdirectory(radmon.fd)
+endif()
+if(REF2TTEN)
+  add_subdirectory(ref2tten.fd)
+endif()
+if(RTMA_MINMAXTRH)
+  add_subdirectory(rtma_minmaxtrh.fd)
+endif()
+if(UPDATE_BC)
+  add_subdirectory(update_bc.fd)
+endif()
+if(UPDATE_GVF)
+  add_subdirectory(update_gvf.fd)
+endif()
+if(UPDATE_ICE)
+  add_subdirectory(update_ice.fd)
+endif()
+if(UPDATE_SNOW)
+  add_subdirectory(update_snow.fd)
+endif()
+if(UPDATE_SST)
+  add_subdirectory(update_sst.fd)
+endif()
+if(USE_RAPHRRR_SFC)
+  add_subdirectory(use_raphrrr_sfc.fd)
+endif()
```

Comment:
@chan-hoo Thanks a lot for making these changes!! A function we've been waiting for!

I have a comment to add a top-level option such as "RRFS_UTL_ALL" which can turn on/off building all executables except those explicitly set in the cmake command line. 
Changes are in this PR to your fork:
https://github.com/chan-hoo/rrfs_utl/pull/1

Reply:
@guoqing-noaa, thanks for the PR. Merged.

Reply:
@chan-hoo Thanks for merging the changes!